### PR TITLE
Add nalgebra 0.33 support

### PR DESCRIPTION
**Pull request by 0e4ef622**
Since the `Allocator` trait changed, I added a trait alias to mimic the old `Allocator` trait.

### CODE REVIEW

**Comment by stefan-k** [comment]
Hi, thanks also for this PR! Would you mind rebasing now that #504 is merged?

File: crates/argmin-math/Cargo.toml [hunk 1]
```diff
@@ -15,11 +15,18 @@ exclude = []
...
 nalgebra_0_32 = { package = "nalgebra", version = "0.32", optional = true }
 nalgebra_0_31 = { package = "nalgebra", version = "0.31", optional = true }
 nalgebra_0_30 = { package = "nalgebra", version = "0.30", optional = true }
 nalgebra_0_29 = { package = "nalgebra", version = "0.29", optional = true }
 
+# simba
+simba_0_6 = { package = "simba", version = "0.6", optional = true }
+simba_0_7 = { package = "simba", version = "0.7", optional = true }
+simba_0_8 = { package = "simba", version = "0.8", optional = true }
+simba_0_9 = { package = "simba", version = "0.9", optional = true }
```
Comment (stefan-k) [inline crates/argmin-math/Cargo.toml:28]: I'm not an nalgebra expert, could you explain to me why simba is needed as a dependency?

File: crates/argmin-math/src/lib.rs [hunk 2]
```diff
@@ -131,14 +132,69 @@
...
+        {}
+        impl<T, R1, C1, R2, C2, U> SameShapeAllocator<T, R1, C1, R2, C2> for U
+        where
+            U: nalgebra::allocator::SameShapeAllocator<R1, C1, R2, C2>,
+            R1: nalgebra::Dim,
+            C1: nalgebra::Dim,
+            R2: nalgebra::Dim,
+            C2: nalgebra::Dim,
+            nalgebra::constraint::ShapeConstraint: nalgebra::constraint::SameNumberOfRows<R1,R2> + nalgebra::constraint::SameNumberOfColumns<C1,C2>,
+        {}
```
Comment (stefan-k) [inline crates/argmin-math/src/lib.rs:164]: I will have to trust you on this ;)

**Review by stefan-k — COMMENTED** [review]
Thanks a lot for this work, this is highly valuable! :) I have one question out of curiosity, otherwise this is good to go!

File: crates/argmin-math/Cargo.toml (hunk 1 above)
Comment (0e4ef622) [reply crates/argmin-math/Cargo.toml:28]: simba is already a dependency of nalgebra, it's just that nalgebra 0.33 no longer re-exports certain traits used in this crate like `simba::scalar::ClosedAdd`.

File: crates/argmin-math/src/lib.rs (hunk 2 above)
Comment (0e4ef622) [reply crates/argmin-math/src/lib.rs:164]: It's mostly copy pasted trait bounds from nalgebra :P

File: crates/argmin-math/Cargo.toml (hunk 1 above)
Comment (0e4ef622) [reply crates/argmin-math/Cargo.toml:28]: Actually taking a second look, I'm just replacing `ClosedAdd` with `ClosedAddAssign` which is reexported in 0.33, so I could remove simba and use the trait from nalgebra instead.

Comment (stefan-k) [reply crates/argmin-math/Cargo.toml:28]: If its not necessary I'd prefer it if it were removed, simply to avoid having to manage it.

**Comment by stefan-k** [comment]
Could you add testing the 0.33 version to the CI, please? 
This would be the appropriate place: 
https://github.com/argmin-rs/argmin/blob/be2444ed875190f361be7cc95a39e63a0c951e68/.github/workflows/ci.yml#L120-L130

File: crates/argmin-math/Cargo.toml (hunk 1 above)
Comment (0e4ef622) [reply crates/argmin-math/Cargo.toml]: Ok, I've removed it.

**Comment by AdrianEddy** [comment]
any update?

**Comment by david-cattermole** [comment]
Hi @stefan-k, it appears that @0e4ef622 added CI support in https://github.com/0e4ef622/argmin/commit/168d918a682b9c13ce69d0e76df02ac4e533aa8f.

**Comment by stefan-k** [comment]
Apologies for being so absent recently. I'll try to address some of the open pull requests on Sunday.

**Comment by david-cattermole** [comment]
> Apologies for being so absent recently. I'll try to address some of the open pull requests on Sunday.

No problem at all, we all get busy! 😄 Thank you for everything you do! 
`argmin` is fantastic, and a lot of people really admire and appreciate your hard-work (I know I do).

**Comment by stefan-k** [comment]
> No problem at all, we all get busy! 😄 Thank you for everything you do! `argmin` is fantastic, and a lot of people really admire and appreciate your hard-work (I know I do).

Thanks for the kind words! :) 
LGTM, Huge thanks to @0e4ef622 for your work and for addressing my request! :)

**Review by stefan-k — APPROVED** [review]

**Pull request merged**